docs/testplan/DWRR-Snappi-Test-Plan.md

```diff
@@ -0,0 +1,797 @@
+   - Flow 4: 25 Gbps with DSCP 4 and sequence numbers
+4. Start traffic and capture on the egress port
+5. Monitor traffic statistics (throughput, packet loss) in real-time during the test
+6. Gather queue statistics from the switch at 60-second intervals
+7. After 180 seconds, stop traffic and capture
+8. Analyze captured traffic to verify sequence integrity and packet counts
+
+#### 2.8.5 Expected Results
+- All packets should be forwarded without any loss (zero packet loss)
+- Sequence numbers should be preserved for each flow (no reordering)
+- Traffic rate at the egress port should match the ingress traffic rate (97G)
+- Queue statistics should accurately reflect the traffic through each queue
+- Each queue should receive and transmit its corresponding DSCP traffic:
+  - Queue 1: ~24 Gbps of DSCP 1 traffic
+  - Queue 2: ~24 Gbps of DSCP 2 traffic
+  - Queue 3: ~24 Gbps of DSCP 3 traffic
+  - Queue 4: ~25 Gbps of DSCP 4 traffic
+
+#### 2.8.6 Success Criteria
+- **Zero Packet Loss**: 100% of transmitted packets must be received (packet count matching between transmit and receive sides)
+- **No Packet Reordering**: All packets must be received in the same sequence they were sent (verified via sequence numbers)
+- **Traffic Rate Accuracy**: The measured egress rate must be within 0.5% of the configured rate (96.5-97.5 Gbps)
+- **Queue Distribution Accuracy**: The measured traffic through each queue must match the configured traffic pattern within 1% tolerance:
+  - Queue 1: 23.5-24.5 Gbps
+  - Queue 2: 23.5-24.5 Gbps
+  - Queue 3: 23.5-24.5 Gbps
+  - Queue 4: 24.5-25.5 Gbps
+- **Switch Statistics Accuracy**: Queue counters on the switch must match the actual traffic within 1% tolerance
+
+## 3. Implementation Details
+
+### 3.1 Snappi Test Framework Implementation
+
+The test scenarios will be implemented using the snappi API. Implementation details will be crafted to generate traffic with appropriate DSCP values and analyze the captured traffic.
+
+### 3.2 Packet Capture Analysis
+
+The captured packets will be analyzed to validate DWRR functionality. The analysis will:
+
+1. Filter packets based on DSCP values
+2. Count the number of packets (or bytes) for each DSCP value
+3. Calculate the percentage distribution
+4. Compare with expected distribution based on configured weights
+5. Determine pass/fail status based on the 2% tolerance criteria
+
+## 4. Test Execution Plan
+
+### 4.1 Test Execution Sequence
+
+Tests should be executed in the following order:
+
+1. Scenario 1: Basic DWRR Validation with 2 Queues
+2. Scenario 2: DWRR with Extreme Weight Ratios
+3. Scenario 3: DWRR Validation with 3 Queues
+4. Scenario 4: DWRR Validation with 4 Queues and Different Traffic Patterns
+5. Scenario 5: DWRR Validation with Different DSCP Distributions on Each Port
+6. Scenario 6: Round-Robin DSCP Pattern Testing
+7. Scenario 7: DWRR Under Varying Congestion Conditions
+8. Scenario 8: Zero-Loss Verification with DWRR Enabled
```

Comment:
I think we should add a test case where it checks the configuration that switch is configured already.

Basically merge Mei's test plan here: https://github.com/sonic-net/sonic-mgmt/pull/16761/files.

No point to add 2 test plans.